### ISSUE TITLE
0 training loss

### ISSUE THREAD

**Issue by santosh-b** [MadryLab/robustness#83]
### For Pytorch 1.7 only:
Hi. I'm getting a `UserWarning: Failed to calculate the accuracy.` which results in 0 loss for training (so- presumably, an error being thrown during gradient update or something). I'm running the following command (from the examples) on Colab: 
```
python -m robustness.main --dataset cifar --data /path/to/cifar \
   --adv-train 0 --arch resnet18 --out-dir /logs/checkpoints/dir/
```
Any help? Thanks

**Comment by Icxa** [comment]
I am encountering the same warning since a couple of days. A script that successfully calculated the accuracy few weeks back, is now giving this UserWarning. 
P.S: I had re-installed the package robustness few days ago.

**Comment by andrewilyas** [comment]
Thanks for bringing this up, looking into it right now. The good thing is that this is an error thrown during calculating the accuracy from the predictions, not from the actual gradient update, so the network should indeed be training. I am looking into the error now. 

Can you tell me what version of PyTorch and robustness you are running?

**Comment by santosh-b** [comment]
Thanks for the quick update. You're absolutely correct, it's a version issue. Pytorch 1.7 causes this to not report accuracy, likely due to some compatibility error within the metrics reporting in the try/catch block. I'll update the original issue accordingly

**Comment by Icxa** [comment]
Yes I confirm the same that it works completely fine with PyTorch 1.6.0. Thank you very much for your response @andrewilyas :)

**Comment by andrewilyas** [comment]
Just to provide an update on this: I've figured out what's changed between 1.6 and 1.7 that causes this error, and will push a hotfix in the next few days. In the meantime, if you want to change your own local copy, it just requires changing ``view`` to ``reshape`` on this line:

https://github.com/MadryLab/robustness/blob/2dabf3bdd8057fdc0718b2f8d8d90d89b1a109df/robustness/tools/helpers.py#L75

That should suffice to fix the issue.

**Comment by andrewilyas** [comment]
This has now been fixed and pushed to PyPI, closing now! Feel free to open a new issue if anything else arises.
